src/mtcp/mtcp_restart.c

```diff
@@ -850,7 +850,10 @@ unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
       mtcp_abort();
     }
     MTCP_ASSERT(vvar == vvarStart);
-    mtcp_memcpy(vvarStart, rinfo->vvarStart, vvarEnd - vvarStart);
+    // On i386, only the first 4096 bytes are readable. Reading beyond that
+    // results in a bus error.
+    // TODO: Find a way to automatically detect the readable bytes.
+    mtcp_memcpy(vvarStart, rinfo->vvarStart, 4096);//vvarEnd - vvarStart);
```

Comment:
I confirm that I still experience bus errors with some 32-bit binaries when multiarch is used.